### DISCUSSION TITLE
how to insert into pair dependent table once; for many of records using prisma

### DISCUSSION THREAD

**Discussion by Abd-elrahman2003** [prisma/prisma#24514] [Q&A]
### Question
```ts
async createIssue(dto:issueDTO){
        let transactions = await this.transactionStoacking(dto)
        let order = await this.prisma.requestOrder.findFirst({where:{id:+dto.issue.requestOrderId}})
        if(!order) throw new HttpException('request order not exists', 400)

        return await this.prisma.$transaction(async prisma=>{
            let issue = await prisma.issueTransfer.create({data:dto.issue})
            const stockTransactionsData = transactions.map((val) => ({
                ...val,
                issueTransactions: {
                create: {
                    issueId: issue.id,
                },
                },
            }));
            
              // Execute createMany in one request
            await prisma.stoackTransaction.createMany({
                data: stockTransactionsData,
                
            });
            return issue
        })
    }
 ```
    here i need to insert into stock transaction table then i want to create stock transfer for this transaction how i can do this using just single query 
    
    look this code is have a problem that he tell me stockTransaction doesn't   have a issueTransaction table ;
    it already exists when i am using just create but with createMany it not working because prisma not supporting
    nested many query
    
    i need any creative idea this will may solve a lot of problem inside my code
 
#23509

### How to reproduce (optional)

create- insert - create many nested loop

### Expected behavior (optional)

createnested query

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it
```prisma
model issueTransactions {
  transactionId Int
  issueId    Int
  transaction   stoackTransaction @relation(fields: [transactionId], references: [id])
  issue         issueTransfer    @relation(fields: [issueId], references: [id])

  @@id([transactionId, issueId])
}


model stoackTransaction {
  id        Int      @id @default(autoincrement())
  type      String
  itemId    Int
  quantity  Float
  price     Float
  fromId    Int
  toId      Int
  netPrice  Float
  createdAt DateTime @default(now())
  item      Item     @relation(fields: [itemId], references: [id])
  from      store    @relation("from", fields: [fromId], references: [id], onDelete: NoAction, onUpdate: NoAction)
  to        store    @relation("to", fields: [toId], references: [id], onDelete: NoAction, onUpdate: NoAction)

  transferTransactions transferTransactions[]
  issueTransactions    issueTransactions[]

}
```

**Comment by ludralph** [comment]
Hi @Abd-elrahman2003 👋 

 One way to go about this is to use a loop to create each `stoackTransaction` individually within the transaction. This way, you can still maintain the atomicity of the transaction while performing the nested create operations. we loop through each transaction and create them individually. This allows you to perform the nested create operation for each `stoackTransaction`.
```ts
async createIssue(dto: issueDTO) {
    let transactions = await this.transactionStoacking(dto);
    let order = await this.prisma.requestOrder.findFirst({ where: { id: +dto.issue.requestOrderId } });
    if (!order) throw new HttpException('request order not exists', 400);

    return await this.prisma.$transaction(async prisma => {
        let issue = await prisma.issueTransfer.create({ data: dto.issue });

        for (const transactionData of transactions) {
            await prisma.stoackTransaction.create({
                data: {
                    ...transactionData,
                    issueTransactions: {
                        create: {
                            issueId: issue.id,
                        },
                    },
                },
            });
        }

        return issue;
    });
}
```

**Reply by Abd-elrahman2003** [reply]
is that not bad for performance make a lot of query requests

**Reply by ludralph** [reply]
Hi @Abd-elrahman2003 

Nested writes using create can be slow, especially when dealing with a large number of records. Each create operation generates a separate database query, which can lead to performance issues. You could consider using createMany instead of multiple create operations.
```js
async createIssue(dto: issueDTO) {
    let transactions = await this.transactionStoacking(dto);
    let order = await this.prisma.requestOrder.findFirst({ where: { id: +dto.issue.requestOrderId } });
    if (!order) throw new HttpException('request order not exists', 400);

    return await this.prisma.$transaction(async prisma => {
        let issue = await prisma.issueTransfer.create({ data: dto.issue });

        // Use createMany instead of multiple create operations
        await prisma.stoackTransaction.createMany({
            data: transactions.map(transactionData => ({
                ...transactionData,
                issueTransactions: {
                    create: {
                        issueId: issue.id,
                    },
                },
            })),
        });

        return issue;
    });
}
```
This approach uses `createMany` to insert all transactions in a single operation, which should be more efficient than multiple individual `create` operations.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!;
